From 14c23a1d291b574f08a386cf32757b320f2db9ca Mon Sep 17 00:00:00 2001
From: Shreenidhi Shedi <sshedi@vmware.com>
Date: Fri, 9 Jul 2021 20:55:00 +0530
Subject: [PATCH] Do not use fallback net config if network files are already
 present

cloud-init generates a fallback net cfg, if no DS found or if a DS is
found and there is no network cfg in DS metadata.

As a result Photon's default network files will be overridden which is
not acceptable.

This new setting will address the issue by checking for
use_fallback_netcfg in cloud.cfg.

Signed-off-by: Shreenidhi Shedi <sshedi@vmware.com>
---
 cloudinit/net/__init__.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/cloudinit/net/__init__.py b/cloudinit/net/__init__.py
index 6b3b84f7..3f6e9ecd 100644
--- a/cloudinit/net/__init__.py
+++ b/cloudinit/net/__init__.py
@@ -16,6 +16,7 @@ from cloudinit import subp
 from cloudinit import util
 from cloudinit.net.network_state import mask_to_net_prefix
 from cloudinit.url_helper import UrlError, readurl
+from cloudinit.settings import CLOUD_CONFIG
 
 LOG = logging.getLogger(__name__)
 SYS_CLASS_NET = "/sys/class/net/"
@@ -490,6 +491,14 @@ def find_fallback_nic_on_linux(blacklist_drivers=None):
 
 def generate_fallback_config(blacklist_drivers=None, config_driver=None):
     """Generate network cfg v2 for dhcp on the NIC most likely connected."""
+
+    key = 'use_fallback_netcfg'
+    cfg = util.read_conf(CLOUD_CONFIG)
+    if key in cfg:
+        LOG.info('%s value is: %s', key, cfg[key])
+        if not cfg[key]:
+            return None
+
     if not config_driver:
         config_driver = False
 
-- 
2.32.0

